rfcs/text/0000-remove-process-ppid.md

```diff
@@ -0,0 +1,136 @@
+-->
+
+## Usage
+
+<!--
+Stage 1: Describe at a high-level how these field changes will be used in practice. Real world examples are encouraged. The goal here is to understand how people would leverage these fields to gain insights or solve problems. ~1-3 paragraphs.
+-->
+
+New processes are typically spawned directly from their parent, or calling, process. Capturing the parent pid (PPID) has many applications:
+
+* Modeling process hierarchy
+* Attackers may try to start a process with an arbitrary parent process set. Capturing the PPID value helps identify if an attacker is attempting privilege escalation through PPID spoofing.
+* Collecting PPID as a possible datapoint to help with the observability of a system.
+
+Users will still be able to capture the PPID in the `process.parent.pid` field. Having the one single field available should hep improve the experience for anyone trying to capture PPIDs from their events.
+
+## Source data
+
+<!--
+Stage 1: Provide a high-level description of example sources of data. This does not yet need to be a concrete example of a source document, but instead can simply describe a potential source (e.g. nginx access log). This will ultimately be fleshed out to include literal source examples in a future stage. The goal here is to identify practical sources for these fields in the real world. ~1-3 sentences or unordered list.
+-->
+
+`process.ppid` is actively populated across several data sources:
+
+* APM Server and agents
+* Beats modules
+* Beats `add_process_metadata` processor
+* Elastic agent integration packages
+
+<!--
+Stage 2: Included a real world example source document. Ideally this example comes from the source(s) identified in stage 1. If not, it should replace them. The goal here is to validate the utility of these field changes in the context of a real world example. Format with the source name as a ### header and the example document in a GitHub code block with json formatting.
+-->
+
+<!--
+Stage 3: Add more real world example source documents so we have at least 2 total, but ideally 3. Format as described in stage 2.
+-->
+
+## Scope of impact
+
+<!--
+Stage 2: Identifies scope of impact of changes. Are breaking changes required? Should deprecation strategies be adopted? Will significant refactoring be involved? Break the impact down into:
+ * Ingestion mechanisms (e.g. beats/logstash)
+ * Usage mechanisms (e.g. Kibana applications, detections)
+ * ECS project (e.g. docs, tooling)
+The goal here is to research and understand the impact of these changes on users in the community and development teams across Elastic. 2-5 sentences each.
+-->
+
+### Ingestion mechanisms
+
+APM, Beats, Elastic Agent, and any processors that populate `process.ppid` today will need to be identified and a migration plan to `process.parent.pid` established.
```

Comment:
Will that a part of this RFC's process, or is that a separate effort?

Reply:
Separate effort; updating those data sources will happen outside the RFC process.

Here we're noting the Elastic data sources that may be impacted when the field is removed from ECS. As this RFC advances, we'll continue communicating that this change is coming, especially to the teams/areas most impacted. The teams can then plan and implement any necessary changes.